lib/active_merchant/billing/gateway.rb

```diff
@@ -209,6 +209,20 @@ def add_field_to_post_if_present(post, options, field)
         post[field] = options[field] if options[field]
       end
 
+      def add_3ds_2_optional_fields(post, options)
+        three_ds_2 = options[:three_ds_2] || {}
+
+        if three_ds_2.has_key?(:optional)
+          three_ds_2[:optional].each do |key, value|
+            normalized_value = normalize(value)
+            next if normalized_value.nil?
+            post[key] = normalized_value unless post[key]
+          end
+        end
+
+        post
+      end
```

Comment:
neat 😎 I guess technically anyone could end up using this and adding top level fields to the post if they knew to add this param; are there any risks there? should we maybe require that they also have included options[:execute_threed]? maybe not, just curious

Reply:
I didn't want to over complicate the logic here. three_ds_2 options come in with execute_threed, so I didn't see too much of a risk when I was implementing. I had considered the change but elected not to do so. That all being said, if someone has a strong opinion on the matter, I'm happy to make the change.